File: nxflix-agents-ts/src/models/grammar.ts
import { z } from 'zod';

export const GrammarCategory = z.enum([
  'formal',
  'classical',
  'conjunctive',
  'conditional',
  'comparative',
  'emphasis',
  'negative',
  'temporal',
  'causative',
  'other',
]);
export type GrammarCategory = z.infer<typeof GrammarCategory>;

export const GrammarPoint = z.object({
  id: z.string(),
  pattern: z.string(),
  meaning: z.string(),
  meaningJp: z.string().nullish(),
  example: z.string(),
  exampleTranslation: z.string(),
  category: GrammarCategory,
  level: z.string().default('N1'),
  notes: z.string().nullish(),
  relatedPatterns: z.array(z.string()).default([]),
});
export type GrammarPoint = z.infer<typeof GrammarPoint>;

File: nxflix-agents-python/src/nxflix_agents/models/grammar.py
"""Grammar point models."""

from enum import Enum
from pydantic import BaseModel


class GrammarCategory(str, Enum):
    """JLPT N1 grammar categories."""

    FORMAL = "formal"
    CLASSICAL = "classical"
    CONJUNCTIVE = "conjunctive"
    CONDITIONAL = "conditional"
    COMPARATIVE = "comparative"
    EMPHASIS = "emphasis"
    NEGATIVE = "negative"
    TEMPORAL = "temporal"
    CAUSATIVE = "causative"
    OTHER = "other"


class GrammarPoint(BaseModel):
    """A JLPT N1 grammar pattern."""

    id: str
    pattern: str
    meaning: str
    meaning_jp: str | None = None
    example: str
    example_translation: str
    category: GrammarCategory
    level: str = "N1"
    notes: str | None = None
    related_patterns: list[str] = []
